src/replica/DatabaseMySQL.h

```diff
@@ -538,15 +538,27 @@ class Connection : public std::enable_shared_from_this<Connection> {
 
     /**
      * Retrieve warning, errors or notes generated after executing the last statement.
-     * This operation is implemented as MySQL query "SHOW WARNINGS".
+     * This operation is implemented using the corresponding MySQL query.
+     * @code
+     *   SHOW WARNINGS [LIMIT <max-num-warnings> [OFFSET <offset>]]
+     * @endcode
      *
      * @note The operation will not reset the query statistics.  Calling this method
      *   many times in a row will return the same result.
      * @see Connection::warningCount
-     *
+     * @param maxNumWarnings The maximum number of entries to be reported. In case
+     *   if both values of the parameter are set to 0 then all entries will be reported.
+     * @param offset The number of entries to skip.
      * @return A collection of entries.
      */
-    std::list<Warning> warnings();
+    std::list<Warning> warnings(unsigned int maxNumWarnings = 0, unsigned int offset = 0);
```

Comment:
The documentation for this parameter incorrectly says that:
```
     * @param maxNumWarnings The maximum number of entries to be reported. In case
     *   if both values of the parameter are set to 0 then all entries will be reported.
```
In reality, if 0 is passed as a value of the parameter, the implementation of the method will use the following MySQL query:
```
SELECT WARNINGS;
```
Where the number of the returned rows depends on a value of the following (GLOBAL or SESSION) MySQL variable:
```
max_error_count
```
The default value of the variable is *64* (and it can't exceed * 65535*) as explained in https://mariadb.com/docs/reference/mdb/system-variables/max_error_count/
If the parameter's value is not *0* then:
```
SELECT WARNINGS LIMIT <maxNumWarnings> [OFFSET <offset>]>;
```
I will rewrite the documentation of the method to correct this issue.